### DISCUSSION TITLE
Phase = -1    [notIndexed]

### DISCUSSION THREAD

**Discussion by zmichels** [mtex-toolbox/mtex#1708] [Ask Anything]
Hi all,

I recently switched to an EDAX acquisition system and have been working with .ang files in MTEX since. When I craft an import script using the import_wizard, everything looks normal as before with .ctf files... but after importing, my "notIndexed" points are assigned a phase of -1, whereas I am used to it being 0... and with all indexed phases having positive integers. However, with the .ang files... now my first indexed phase has a "phase" value of 0. This is surmountable, but annoying. I have a number of scripts that anticipate indexed phases have "phase" values > 0.

Any thoughts on how I might adapt the import script or something to yield a result similar to what I get when I process CTF files?

Thanks for any insights.

Zach

![image](https://user-images.githubusercontent.com/8769619/211676475-7490c051-338a-4aed-9bc6-9972068b84b6.jpeg)

**Comment by kilir** [comment]
Hi Zach,

I can understand this. The -1 stems from the fact that .ang files use a different concept of marking non-indexed points (and it also depends which software/version wrote the file as well as whether it's single or multiphase; in the latter it's 0). Here's a note https://github.com/mtex-toolbox/mtex/blob/600a2504b4cb09ff939e26cfe57373c783386088/interfaces/loadEBSD_ang.m#L163

However, having it consistent would be something which could be fixed one moment. In the meantime, doesn't `~(ebsd.isIndexed)` produce not indexed pixels without relying on the phase number?

Cheers,
Rüdiger

**Reply by zmichels** [reply]
That does yield a vector that needing which EBSD data belong to "notIndexed" phase. Thanks for that.

I understand what you're saying, and yet I feel a phase of -1 is nonsensical, let alone a coding nuisance due to inconsistency as you mentioned.

**Comment by Hyperion1Q84** [comment]
so how to fix this problem？
I want to change to Phase = 0 [notIndexed]

**Reply by kilir** [reply]
Hi,

`ebsd.phaseMap = [0 ;1]` should do.

Cheers,
Rüdiger

**Reply by Hyperion1Q84** [reply]
Hi，

There is another problem when ` ebsd.phaseMap = [0 ;1]`

`>> ebsd.phaseMap = [0 ;1]
 
ebsd = EBSD (y↑→x)
 
错误使用 assert
The second input argument can be a map of RGB values (size Nx3).

出错 colornames>cnClosest (第 353 行)
assert(ismatrix(inp)&&size(inp,2)==3&&isreal(inp)&&all(0<=inp(:)&inp(:)<=1),...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
出错 colornames (第 217 行)
	idx = cnClosest(dtE,ord,rgb,varargin{1},varargin(2:end));
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
出错 rgb2str (第 10 行)
  str = char(colornames(getMTEXpref('colorPalette','CSS'),rgb));
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
出错 EBSD/display (第 38 行)
  matrix{ip,4} = rgb2str(CS.color);
                 ^^^^^^^^^^^^^^^^^`

**Reply by ralfHielscher** [reply]
I would not call this a problem. The point is that the phase is exported in the |ang| file as -1 and 0. Hence MTEX import them as such. Could you please write what you want to achieve an why it is a problem, that the not indexed phase is -1?

Ralf.

**Reply by Hyperion1Q84** [reply]
Hello, 
I want to calculate the GND dislocation density through additional python code published on Github: "https://github.com/tskippon/GND". 
Specify unindexed phase = 0 in this code. 
I tried to modify the code, but it didn't work, that's why I thought of changing unindexed phase =-1 to unindexed phase = 0 during the import phase of MTEX 
Thank you very much for your help 
Thank you!
